projects/igniteui-angular/src/lib/data-operations/filtering-strategy.ts

```diff
@@ -40,7 +40,9 @@ export abstract class BaseFilteringStrategy implements IFilteringStrategy  {
     // protected
     public findMatchByExpression(rec: any, expr: IFilteringExpression, isDate?: boolean, isTime?: boolean, grid?: GridType): boolean {
         const val = this.getFieldValue(rec, expr.fieldName, isDate, isTime, grid);
-        return expr.condition?.logic(val, expr.searchVal, expr.ignoreCase);
+        if (expr.condition.logic) {
```

Comment:
Well, if there's no condition, this would crash. Maybe you meant something like:
```suggestion
        if (expr.condition?.logic) {
```

Reply:
done, thanks